spec/models/manageiq/providers/hawkular/middleware_manager_spec.rb

```diff
@@ -6,4 +6,49 @@
   it ".description" do
     expect(described_class.description).to eq('Hawkular')
   end
+
+  describe "miq_id_prefix" do
+    let(:random_id) { SecureRandom.hex(10) }
+    let(:my_region) do
+      MiqRegion.my_region || FactoryGirl.create(:miq_region, :region => MiqRegion.my_region_number)
+    end
+    let(:random_region) do
+      region = Random.rand(1..99) while !region || region == my_region.region
+      MiqRegion.find_by(:region => region) || FactoryGirl.create(:miq_region, :region => region)
+    end
+
+    before(:each) do
+      # Ensure "my region" exists
+      my_region
+    end
+
+    it "must return non-empty string" do
+      rval = subject.miq_id_prefix
+
+      expect(rval).to be_a(String)
```

Comment:
We don't test types in Ruby unless strictly necessary, and specially not on primitives.